entry/src/entry.rs

```diff
@@ -111,15 +111,22 @@ pub struct Api<'a> {
 /// field points to Transactions that took place shortly before `hash` was generated.
 ///
 /// If you multiply `num_hashes` by the amount of time it takes to generate a new hash, you
-/// get a duration estimate since the last Entry. Since processing power increases
+/// get a duration estimate since the last `Entry`. Since processing power increases
 /// over time, one should expect the duration `num_hashes` represents to decrease proportionally.
 /// An upper bound on Duration can be estimated by assuming each hash was generated by the
 /// world's fastest processor at the time the entry was recorded. Or said another way, it
 /// is physically not possible for a shorter duration to have occurred if one assumes the
 /// hash was computed by the world's fastest processor at that time. The hash chain is both
 /// a Verifiable Delay Function (VDF) and a Proof of Work (not to be confused with Proof of
 /// Work consensus!)
-
+///
+/// `Entry` contains `transactions` that are executable in parallel. So an `Entry` is a unit of
+/// parallelism at the chain level. All transactions in the `transactions` field have to follow the
+/// read/write locking restrictions with regard to the accounts they reference. A single account can
+/// be either written by a single transaction, or read by one or more transactions, but not both.
+///
+/// This enforcement is done via a call to `solana_runtime::accounts::Accounts::lock_accounts()`
+/// with the `txs` argument holding all the `transactions` in the `Entry`.
```

Comment:
There are two places where this is enforced.  One in TPU and one in TVU.

Your clarification specifies the TPU path, but then the TVU path is missing.
TVU calls it like this:

```rust
solana_ledger::blockstore_processor::confirm_slot_entries()
solana_ledger::blockstore_processor::process_entries_with_callback()
solana_runtime::bank::Bank::prepare_sanitized_batch()
solana_runtime::accounts::Accounts::lock_accounts()
```

The reason I did not specify it more precisely, is because there are two paths.  And I did not want too much stuff from the code in the comment.
I figured, one can follow references to `solana_runtime::accounts::Accounts::lock_accounts()` to find both locations.

Do you still want to just apply your clarification on the TPU case?